### PR TITLE
Fix cases of scroll issues with positionFixed enabled

### PR DESCRIPTION
**Pull request by MrFrankel**

### CODE REVIEW

File: packages/popper/tests/functional/core.js [hunk 1]
```diff
@@ -115,10 +115,10 @@ const arrowSize = 5;
       prepend(reference, document.body);
       prepend(popper, document.body);
 
-      new Popper(reference, popper, {
-        onCreate(data) {
-          const bottom = getRect(popper).bottom;
-          expect(bottom).toBeApprox(getRect(reference).top);
+    new Popper(reference, popper, {
+      positionFixed: false,onCreate(data) {
```
Comment (FezVrasta) [inline packages/popper/tests/functional/core.js]: Why?

File: packages/popper/src/utils/getOffsetRectRelativeToArbitraryNode.js [hunk 2]
```diff
@@ -16,6 +16,10 @@ export default function getOffsetRectRelativeToArbitraryNode(children, parent) {
   const borderTopWidth = parseFloat(styles.borderTopWidth, 10);
   const borderLeftWidth = parseFloat(styles.borderLeftWidth, 10);
 
+  if(ignoreNegativeParentScroll) {
+    parentRect.top = parentRect.top > 0 ? parentRect.top : 0;
```
Comment (FezVrasta) [inline packages/popper/src/utils/getOffsetRectRelativeToArbitraryNode.js]: `Math.max(0, parentRect.top)`?

**Comment by FezVrasta** [comment]
I'm not sure, but it looks like this thing is getting messy..

File: packages/popper/tests/functional/core.js (hunk 1 above)
Comment (MrFrankel) [reply packages/popper/tests/functional/core.js]: well, it seems like an edge case that only fails on IE.

File: packages/popper/src/utils/getOffsetRectRelativeToArbitraryNode.js (hunk 2 above)
Comment (MrFrankel) [reply packages/popper/src/utils/getOffsetRectRelativeToArbitraryNode.js]: good point

**Comment by MrFrankel** [comment]
Actually, considering the change and the benefit derived from having this option available, I think its rather reasonable, 4-5 special cases only effected with the flag on

File: packages/popper/tests/functional/core.js (hunk 1 above)
Comment (FezVrasta) [reply packages/popper/tests/functional/core.js]: In this case I'd rather have something like `if (positionFixed && !isIE) { /* run test */ }` for just the specific test cases.

File: packages/popper/src/methods/update.js [hunk 3]
```diff
@@ -58,7 +59,7 @@ export default function update() {
   );
   data.offsets.popper.position = this.options.positionFixed ? 'fixed' : 'absolute';
 
-  // run the modifiers
+   // run the modifiers
```
Comment (FezVrasta) [inline packages/popper/src/methods/update.js]: Unneeeded extra space

File: packages/popper/src/methods/update.js [hunk 4]
```diff
@@ -30,7 +30,8 @@ export default function update() {
     this.state,
     this.popper,
     this.reference,
-    this.options.positionFixed ? window.document.documentElement : undefined
+    this.options.positionFixed ? window.document.documentElement : undefined,
```
Comment (FezVrasta) [inline packages/popper/src/methods/update.js]: What's the rationale behind these checks? May you add a variable named with some meaningful name and maybe also add a comment to better explain the thing?

Comment (FezVrasta) [inline packages/popper/src/methods/update.js]: Do we need to pass this as argument if the only needed variable to evaluate this is already passed as argument as well?

File: packages/popper/tests/functional/core.js (hunk 1 above)
Comment (MrFrankel) [reply packages/popper/tests/functional/core.js]: ok, sounds good

File: packages/popper/src/methods/update.js (hunk 3 above)
Comment (MrFrankel) [reply packages/popper/src/methods/update.js]: will fix

File: packages/popper/src/methods/update.js (hunk 4 above)
Comment (MrFrankel) [reply packages/popper/src/methods/update.js]: well, me thought is that the parameter 'fixedParent' could be used in the future for other cases where someone would want to force a parent as a reference.
If I move it inside I force a coupling between setting a specific parent and cases where you want to ignore negative offsets. 

Ill try and see if I can make this more elegant without having 2 parameters

**Comment by MrFrankel** [comment]
I am withdrawing this PR in favor of a small refactor.
I have done some re writing according to some of your comments which I think will be more aligned with the approach of the rest of the code.

Ill make a new PR when i'm finished
